include/constantine/protocols/ethereum_verkle_ipa.h

```diff
@@ -0,0 +1,228 @@
+    cttEthVerkleIpa_EccPointNotInSubGroup
+} ctt_eth_verkle_ipa_status;
+
+static const char* ctt_eth_verkle_ipa_status_to_string(ctt_eth_verkle_ipa_status status) {
+  static const char* const statuses[] = {
+    "cttEthVerkleIpa_Success",
+    "cttEthVerkleIpa_VerificationFailure",
+    "cttEthVerkleIpa_InputsLengthsMismatch",
+    "cttEthVerkleIpa_ScalarZero",
+    "cttEthVerkleIpa_ScalarLargerThanCurveOrder",
+    "cttEthVerkleIpa_EccInvalidEncoding",
+    "cttEthVerkleIpa_EccCoordinateGreaterThanOrEqualModulus",
+    "cttEthVerkleIpa_EccPointNotOnCurve",
+    "cttEthVerkleIpa_EccPointNotInSubGroup"
+  };
+  size_t length = sizeof statuses / sizeof *statuses;
+  if (0 <= status && status < length) {
+    return statuses[status];
+  }
+  return "cttEthVerkleIpa_InvalidStatusCode";
+}
+
+// Opaque types for Nim-defined types
+typedef struct Fr Fr;
+typedef struct Banderwagon Banderwagon;
+typedef struct EC_TwEdw EC_TwEdw;
+typedef struct Fp Fp;
+
+typedef union {
+    uint8_t u8;
+    uint16_t u16;
+    uint32_t u32;
+    uint64_t u64;
+    unsigned int u;
+} SomeUnsignedInt;
+
+typedef struct {
+    SomeUnsignedInt* values;  
+    size_t length;
+} SomeUnsignedInt_Array;
+
+// Define types for openArray
+typedef struct {
+    Fr* data;    
+    size_t len;   
+} Fr_BanderWagon_OpenArray;
+
+typedef struct {
+    EC_TwEdw* data;    
+    size_t len;         
+} EC_TwEdw_Fp_Banderwagon_OpenArray;
+
+typedef struct {
+    byte value[32];  // 32-byte array for field element
+} Fp_Banderwagon;
+
+typedef struct {
+    byte value[32];  // 32-byte array for field element
+} Fr_Banderwagon;
+
+typedef struct {
+    byte x[32];  // 32-byte x-coordinate
+    byte y[32];  // 32-byte y-coordinate
+} EC_TwEdw_Fp_Banderwagon;
+
+typedef struct {
+    byte x[32];  // 32-byte x-coordinate
+    byte y[32];  // 32-byte y-coordinate
+} EC;
+
+typedef struct {
+    EC* points;  
+    size_t length;  
+} PolynomialEval_EC;
+
+typedef struct {
+    Fr* points;  
+    size_t length;  
+} PolynomialEval_Fr;
+
+typedef struct {
+    EC* points;  
+    size_t length;  
+} PolynomialEval_EcAff;
+
+typedef struct {
+    Fr* domain_values;  
+    size_t length; 
+} PolyEvalLinearDomain_Fr;
+
+typedef struct {
+    EC* ec_points;  
+    Fr* field_elements;      
+    size_t logN;                         
+} IpaProof_EcAff_Fr;
+
+typedef struct {
+    EC* ec_points;  
+    Fr* field_elements;      
+    size_t logN;                         
+} IpaMultiProof_EcAff_Fr;
+
+typedef struct {
+    EC* ec_points;  
+    size_t length;                       
+} EcAffArray;
+
+typedef struct EthVerkleIpaProof EthVerkleIpaProof;
+typedef struct EthVerkleIpaMultiProof EthVerkleIpaMultiProof;
+typedef struct IpaProof IpaProof;
+typedef struct IpaMultiProof IpaMultiProof;
+typedef struct EthVerkleTranscript EthVerkleTranscript;
+
+typedef byte EthVerkleIpaProofBytes[544];       // Array of 544 bytes
+typedef byte EthVerkleIpaMultiProofBytes[576]; 
```

Comment:
It's more useful to provide:
- either `ctt_eth_verkle_ipa_proof_sizeof()` and `ctt_eth_verkle_ipa_multiproof_sizeof()` and let the caller do their allocation / destruction
- or `ctt_eth_verkle_ipa_proof_create()` / `ctt_eth_verkle_ipa_proof_destroy()`
  and `ctt_eth_verkle_ipa_multiproof_create()` / `ctt_eth_verkle_ipa_multiproof_destroy()`